### DISCUSSION TITLE
Highly reflective and mirror object don't illuminate surroundings when a spotlight is shone on them.

### DISCUSSION THREAD

**Discussion by buddhasystem** [mitsuba-renderer/mitsuba3#1243] [Q&A] [answered]
Dear All,

I've placed a spotlight at the sensor location i.e. on the front face of the chamber, and it's shining inwards. Inside, there is a disk mirror tilted such that it is expected to reflect the light downwards. Here is the result:
![image](https://github.com/user-attachments/assets/3ea5b4a5-1b0e-4fca-a24b-11de5b6ec764)

When I replace the mirror with a highly reflective "diffuse" material, I get something roughly similar i.e. I don't see extra light on the bottom:
![image](https://github.com/user-attachments/assets/39f8b076-8ffe-49c5-8218-9a6707fb43dd)

Please advise. Thank you.

**Comment by njroussel** [comment]
Hi @buddhasystem 

This is expected, the light paths you're trying to construct are too complex:
The path starts at the camera, hits the bottom floor, reflects toward the mirror, reflects of the mirror, hits the light source.

The last two events are improbable. Due to the Dirac delta nature of the mirror BDSF, you will never hit the light source after having reflected on the mirror. 

The usual fix is more of hack: replace the `conductor` BSDF with a `roughconductor` and give it a small roughness. This will allow light paths to be constructed however improbable they are.

**Comment by buddhasystem** [comment]
@njroussel Many thanks for your helpful comment. I'm unsure of the situation somewhat still, i.e. this geometry is actually quite simple and I expect the floor to get extra illumination. The floor is not mirror, would it not be OK to shine light on it from a mirror and see diffusion? I'm shining a light from the center of the screen, the floor is diffuse, so I am expecting a spot. I think that's the key, the floor is diffuse and not Dirac. In fact the lower picture has a diffuse disk as well and still no sign of any additional light on the bottom.

I appreciate what you are saying about light traveling in the opposite direction, it's reversible.

I've just changed the material of both the reflecting disk and the floor to "bright" which I defined as diffuse with reflectance of 0.9999 -- which I think is a lot, and I still don't see any extra light apparent on the floor:
![image](https://github.com/user-attachments/assets/36609fab-6003-4c29-8ece-dc79b74b388c)

In the actual problem I'm trying to solve I will need to have a few mirrors because that's how my apparatus is built. So using diffuse can be useful for debugging but not for the actual solution. My hope is that the geometry will be different, relative location of the film and the light source will be also different, in fact the light source will be extended in the volume of the chamber -- I'll have to write an emitter plug in for that. That's unrelated to the topic of this thread, though, I'll create another one.

**Reply by buddhasystem** [reply, marked as answer]
I think this is solved, I get a reflection of a diffuse disk on the mirror floor now. It was a matter of moving the objects just to get the right reflection.
